envi/archs/arm/disasm.py

```diff
@@ -2591,9 +2689,29 @@ def repr(self, op):
 
 class ArmDisasm:
     fmt = None
-    def __init__(self, endian=ENDIAN_LSB):
+    #This holds the current running Arm instruction version and mask
+    _archVersionMask = 'ARMv7A'
+
+    def __init__(self, endian=ENDIAN_LSB, mask = 'ARMv7A'):
+        self.setArchMask(mask)
         self.setEndian(endian)
-        
+
+    def setArchMask(self, key = 'ARMv7R'):
+        ''' set arch version mask '''
+        if key in ARCH_REVS:
+            self._archVersionMask = ARCH_REVS[key]
+        elif key == 'thumb16':
```

Comment:
given the current approach to arm/thumb, this will never happen.
'thumb' and 'thumb16' are handed into the ThumbDisasm class, not the ArmDisasm class.  

the problem is, the ThumbModule class will take that name and use it to identify the thumb architecture module, so we can't just hand in the ARMv7 title in the same way.  furthermore, we don't want to.  we need *that* name to stay relatively static ('arm', 'thumb').  we'll want to figure out a way to have these "personalities" or "versions" be specified elsewhere (or as a separate argument to the constructor).

please remove these elifs.